packages/svelte/src/compiler/phases/1-parse/read/style.js

```diff
@@ -414,6 +395,32 @@ function read_declaration(parser) {
 	};
 }
 
+/**
+ * Reads a declaration, rule or at-rule
+ *
+ * @param {import('../index.js').Parser} parser
+ * @returns {import('#compiler').Css.Declaration | import('#compiler').Css.Rule | import('#compiler').Css.Atrule}
+ */
+function read_block_item(parser) {
+	if (parser.match('@')) {
+		return read_at_rule(parser);
+	}
+
+	const start = parser.index;
+	parser.read_until(REGEX_SEMICOLON_OR_OPEN_BRACE_OR_CLOSE_BRACE);
```

Comment:
using regexes for this sort of thing always makes me a bit nervous — have we tested this against e.g. values containing strings that contain special characters? see e.g. #10221